x-pack/plugins/canvas/public/components/workpad_header/control_settings/auto_refresh_controls.js

```diff
@@ -24,111 +21,82 @@ import {
 } from '@elastic/eui';
 import { timeDurationString } from '../../../lib/time_duration';
 import { RefreshControl } from '../refresh_control';
+import { CustomInterval } from './custom_interval';
 
 const ListGroup = ({ children }) => <ul style={{ listStyle: 'none', margin: 0 }}>{[children]}</ul>;
 
-export class AutoRefreshControls extends Component {
-  static propTypes = {
-    refreshInterval: PropTypes.number,
-    setRefresh: PropTypes.func.isRequired,
-    disableInterval: PropTypes.func.isRequired,
-  };
+export const AutoRefreshControls = ({ refreshInterval, setRefresh, disableInterval }) => {
+  const RefreshItem = ({ duration, label }) => (
+    <li>
+      <EuiLink onClick={() => setRefresh(duration)}>{label}</EuiLink>
+    </li>
+  );
 
-  refreshInput = null;
+  return (
+    <div>
+      <EuiFlexGroup alignItems="center" justifyContent="spaceAround" gutterSize="xs">
```

Comment:
This component has a ton of moving parts and a lot of markup.  I'd like to see this component broken up into its disparate parts as separate components with independent stories/snapshots... and Typescript'd.  This component in particular should be fairly straightforward.